test/conditional_check_test.go

```diff
@@ -0,0 +1,96 @@
+// +build e2e
+
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"testing"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	tb "github.com/tektoncd/pipeline/test/builder"
+	knativetest "knative.dev/pkg/test"
+)
+
+func TestPipelineRunConditionalCheck(t *testing.T) {
+	c, namespace := setup(t)
+	t.Parallel()
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(t, c, namespace) }, t.Logf)
+	defer tearDown(t, c, namespace)
+
+	condResource := tb.Condition("file-exists", namespace,
+		tb.ConditionSpec(
+			tb.ConditionParamSpec("path", v1alpha1.ParamTypeString),
+			tb.ConditionResource("workspace", v1alpha1.PipelineResourceTypeGit),
+			tb.ConditionSpecCheck("", "alpine"),
+			tb.ConditionSpecCheckScript("cat $(resources.workspace.path)/$(params.path)| grep \"Cloud Native\""),
```

Comment:
I dunno about this bit - we're kind of assuming that README.md will always contain the words "Cloud Native".  Changing a README file in the future shouldn't result in a e2e test failure and will be super confusing if we ever do.

How about checking whether this very test file exists?  If you want to keep the "grep" then you could add a comment here like `// This comment is used by this test` and then grep for that with `cat $(resources.workspace.path)/$(params.path) | grep \"This comment is used by this test\"`